assignment02/voronoi.c

```diff
@@ -0,0 +1,276 @@
+            }
+        }
+    }
+}
+
+void generate_map_gates(terrain_map_t *map, Gates_t* gates) 
+{
+    // these may seem "backwards" but x gives us row and y gives entry in that row
+    if(gates->left.x == 0 && gates->left.y == 0) {
+        gates->left.x = 1 + (rand() % (GRID_HEIGHT - 2));
+        gates->left.y = 0; 
+    }
+    if(gates->right.x == 0 && gates->right.y == 0) {
+        gates->right.x = 1 + (rand() % (GRID_HEIGHT - 2));
+        gates->right.y = GRID_WIDTH - 1; 
+    }
+    if(gates->bottom.x == 0 && gates->bottom.y == 0) {
+        gates->bottom.x = GRID_HEIGHT - 1;
+        gates->bottom.y = 1 + (rand() % (GRID_WIDTH - 2));
+    }
+    if(gates->top.x == 0 && gates->top.y == 0) {
+        gates->top.x = 0;
+        gates->top.y = 1 + (rand() % (GRID_WIDTH - 2));
+    }
+    map->grid[gates->left.x][gates->left.y]     = pathTile.ascii;
+    map->grid[gates->right.x][gates->right.y]   = pathTile.ascii;
+    map->grid[gates->bottom.x][gates->bottom.y] = pathTile.ascii;
+    map->grid[gates->top.x][gates->top.y]       = pathTile.ascii;
+}
+
+void generate_map_path(terrain_map_t *m, Gates_t gates)
+{
+    int direction;
+
+    
+    // left gate: go right until bottom gate
+    while(gates.left.y < gates.bottom.y) {
+        m->grid[gates.left.x][gates.left.y] = pathTile.ascii;
+        gates.left.y++;
+    }
+
+
+    direction = gates.left.x > gates.right.x ? -1 : 1;
+    // left gate: go in virtical direction of right gate
+    while(gates.left.x != gates.right.x) {
+        m->grid[gates.left.x][gates.left.y] = pathTile.ascii;
+        gates.left.x += direction;
+    }
+    
+
+    // left gate: go rest of way in horizontal to right gate
+    while(gates.left.y != GRID_WIDTH) {
+        m->grid[gates.left.x][gates.left.y] = pathTile.ascii;
+        gates.left.y++;
+    }
+
+
+    // bottom gate: go up until right gate (gates.left.x has been changed)
+    while(gates.bottom.x > gates.left.x) {
+        m->grid[gates.bottom.x][gates.bottom.y] = pathTile.ascii;
+        gates.bottom.x--;
+    }
+
+    
+    // bottom gate: go in horizontal direction of top gate
+    direction = gates.bottom.y > gates.top.y ? -1 : 1;
+    while(gates.bottom.y != gates.top.y) {
+        m->grid[gates.bottom.x][gates.bottom.y] = pathTile.ascii;
+        gates.bottom.y += direction;
+    }
+
+    // bottom gate: go rest of the way up to top gate
+    while(gates.bottom.x != 0) {
+        m->grid[gates.bottom.x][gates.bottom.y] = pathTile.ascii;
+        gates.bottom.x--;
+    }
+
+    if(m->location.y == 400){
+        m->grid[gates.top.x][gates.top.y] = mountainTile.ascii;
+    }
+    if(m->location.x == 0){
+        m->grid[gates.left.x][gates.left.y] = mountainTile.ascii;
+    }
+    if(m->location.y == 0){
+        m->grid[gates.bottom.x][gates.bottom.y] = mountainTile.ascii;
+    }
+    if(m->location.x == 400){
+        m->grid[gates.right.x][gates.right.y] = mountainTile.ascii;
+    }
```

Comment:
I'm pretty sure u mean to make the edge of the map to not have gates but I'm almost certain this doesn't do anything

Reply:
if you run it, this does infact make all edges mountaintiles, thereby "blocking the gates"

Reply:
prolly not best way to do it, but this is kinda temp fix till later implementations i think.